mrp_sale_fields/models/sale_order.py

```diff
@@ -22,10 +22,13 @@ def _compute_production_ids(self):
             (tuple(self.ids),),
         )
         production_data = self._cr.fetchall()
-        if production_data:
-            mapped_data = dict(production_data)
-            for order in self.browse(mapped_data.keys()):
-                order.production_count = mapped_data.get(order.id, 0)
+        if not production_data:
+            for order in self:
+                order.production_count = 0
+            return
```

Comment:
Ensure each sale order has 0 in its production_count fields when it doesn't have related manufacture orders since next loop can't set those orders